File: app/src/main/java/database/AccountDetailsDao.java
package database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

import models.AccountDetails;

    @Dao
    public interface AccountDetailsDao {

        @Query("SELECT * FROM AccountDetails")
        List<AccountDetails> getAll();

        @Insert
        void insertAll(AccountDetails... accountDetails);

        @Update
        int update(AccountDetails accountDetails);

        @Delete
        boolean delete(AccountDetails... accountDetails);
    }

File: app/src/main/java/database/AccountDetailsDao.kt
package database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import models.AccountDetails

@Dao
interface AccountDetailsDao {
    @get:Query("SELECT * FROM AccountDetails")
    val all: List<AccountDetails?>?

    @Query("SELECT * FROM AccountDetails WHERE Id = :id")
    fun getSelectedModel(id: Long): AccountDetails?

    @Query("Delete FROM AccountDetails WHERE Id = :id")
    fun deleteSelectedModel(id: Long): Int

    @Insert
    fun insertAll(vararg accountDetails: AccountDetails?)

    @Update
    fun update(accountDetails: AccountDetails?): Int
}